### PR TITLE
fix: support changing of log level within the tool

### PR DESCRIPTION
**Pull request by realekhor**

### CODE REVIEW

**Comment by tdashworth** [comment]
@realekhor thanks for this. Happy to review this in a few days.

**Comment by realekhor** [comment]
thanks @tdashworth

**Comment by tdashworth** [comment]
@realekhor Could the selection box be moved the to final page? 

image

**Comment by realekhor** [comment]
@tdashworth good point. I will try make the change.

**Comment by realekhor** [comment]
@tdashworth controls now moved to requested pages. hope this is ok.

**Comment by realekhor** [comment]
@tdashworth looks like i need to add more code coverage tests :(

**Comment by realekhor** [comment]
@tdashworth not sure we ever went beyond 75% coverage :). so i think it is ready

**Comment by tdashworth** [comment]
> @tdashworth not sure we ever went beyond 75% coverage :). so i think it is ready

Agreed, our code coverage isn't as high as we'd like. Please could we add tests for these two lines:

- https://sonarcloud.io/code?id=xrm-datamigration-xrmtoolbox&pullRequest=44&selected=xrm-datamigration-xrmtoolbox%3ACapgemini.Xrm.CdsDataMigrator%2FCapgemini.Xrm.CdsDataMigratorLibrary%2FUserControls%2FImportWizard.cs&line=224
- https://sonarcloud.io/code?id=xrm-datamigration-xrmtoolbox&pullRequest=44&selected=xrm-datamigration-xrmtoolbox%3ACapgemini.Xrm.CdsDataMigrator%2FCapgemini.Xrm.CdsDataMigratorLibrary%2FUserControls%2FExportWizard.cs&line=192

**Review by tdashworth — DISMISSED** [review]

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs [hunk 1]
```diff
@@ -92,7 +93,7 @@ public void ExportDataActionThrowsException()
                                 .Throws<OrganizationalServiceException>();
             logger.Setup(a => a.LogError(It.IsAny<string>()));
 
-            FluentActions.Invoking(() => systemUnderTest.ExportDataAction())
+            FluentActions.Invoking(() => systemUnderTest./*ExportDataAction*/ExportData(null, new EventArgs()))
```
Comment (tdashworth) [inline Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs:96]: Can this comment be removed? 
```suggestion
            FluentActions.Invoking(() => systemUnderTest.ExportData(null, new EventArgs()))
```

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs [hunk 2]
```diff
@@ -108,13 +109,38 @@ public void ExportDataActionDoesNotThrowException()
             dataMigrationService.Setup(a => a.ExportData(It.IsAny<ExportSettings>()));
             logger.Setup(a => a.LogError(It.IsAny<string>()));
 
-            FluentActions.Invoking(() => systemUnderTest.ExportDataAction())
+            FluentActions.Invoking(() => systemUnderTest./*ExportDataAction*/ExportData(null, new EventArgs()))
```
Comment (tdashworth) [inline Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs:112]: Can this comment be removed? 
```suggestion
            FluentActions.Invoking(() => systemUnderTest.ExportData(null, new EventArgs()))
```

**Review by tdashworth — CHANGES_REQUESTED** [review]

File: Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs (hunk 2 above)
Comment (realekhor) [reply Capgemini.Xrm.CdsDataMigrator/Capgemini.Xrm.CdsDataMigratorLibrary.Tests.Unit/Presenters/ExportPresenterTests.cs]: addressed in latest commit

**Review by tdashworth — APPROVED** [review]

**Comment by tdashworth** [comment]
@realekhor Thank you so much for the work you've put into this. It's very much appreciated!

**Pull request merged**